OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -1915,95 +1777,35 @@ public static void deleteTags(Collection<String> keys, ChangeTagsUpdateHandler h
-         JSONObject jsonTags = new JSONObject();
-
-         for (int i = 0; i < jsonArray.length(); i++)
-            jsonTags.put(jsonArray.getString(i), "");
-
-         sendTags(jsonTags, handler);
-      } catch (Throwable t) {
-         Log(LOG_LEVEL.ERROR, "Failed to generate JSON for deleteTags.", t);
-      }
-   }
-
-   public static void idsAvailable(IdsAvailableHandler inIdsAvailableHandler) {
-
-      //if applicable, check if the user provided privacy consent
-      if (shouldLogUserPrivacyConsentErrorMessageForMethodName("idsAvailable()"))
-         return;
-
-      idsAvailableHandler = inIdsAvailableHandler;
-
-      Runnable runIdsAvailable = new Runnable() {
-         @Override
-         public void run() {
-            if (getUserId() != null)
-               OSUtils.runOnMainUIThread(new Runnable() {
-                  @Override
-                  public void run() {
-                     internalFireIdsAvailableCallback();
-                  }
-               });
-         }
-      };
-
-      if (appContext == null || shouldRunTaskThroughQueue()) {
-         Log(LOG_LEVEL.ERROR, "You must initialize OneSignal before getting tags! " +
-                 "Moving this tag operation to a pending queue.");
-         addTaskToQueue(new PendingTaskRunnable(runIdsAvailable));
-         return;
-      }
+   }
 
-      runIdsAvailable.run();
+   public static void deleteTags(String jsonArrayString) {
+      deleteTags(jsonArrayString, null);
    }
 
-   static void fireIdsAvailableCallback() {
-      if (idsAvailableHandler != null) {
-         OSUtils.runOnMainUIThread(new Runnable() {
-            @Override
-            public void run() {
-               internalFireIdsAvailableCallback();
-            }
-         });
+   public static void deleteTags(String jsonArrayString, ChangeTagsUpdateHandler handler) {
+      try {
+         deleteTags(new JSONArray(jsonArrayString), handler);
+      } catch (Throwable t) {
+         Log(LOG_LEVEL.ERROR, "Failed to generate JSON for deleteTags.", t);
       }
    }
 
-   private synchronized static void internalFireIdsAvailableCallback() {
-      if (idsAvailableHandler == null)
+   public static void deleteTags(JSONArray jsonArray, ChangeTagsUpdateHandler handler) {
+      //if applicable, check if the user provided privacy consent
+      if (shouldLogUserPrivacyConsentErrorMessageForMethodName("deleteTags()"))
          return;
 
-      String regId = OneSignalStateSynchronizer.getRegistrationId();
-      if (!OneSignalStateSynchronizer.getSubscribed())
-         regId = null;
-
-      String userId = getUserId();
-      if (userId == null)
-         return;
+      try {
+         JSONObject jsonTags = new JSONObject();
 
-      idsAvailableHandler.idsAvailable(userId, regId);
+         for (int i = 0; i < jsonArray.length(); i++)
+            jsonTags.put(jsonArray.getString(i), "");
 
-      if (regId != null)
-         idsAvailableHandler = null;
+         sendTags(jsonTags, handler);
```

Comment:
we should log something here to make clear why deleting tags results in a send tags call